batch-setup/make_meta_tiles.py

```diff
@@ -137,31 +141,27 @@ def read_metas_to_file(self, filename, present=False, compress=False):
                '-max-zoom', str(self.max_zoom),
                stdout=filename)
 
-    def generate_missing_tile_coords(self, try_generator=False):
-        # type: () -> List[Coordinate]
+    def generate_missing_tile_coords(self, tmpdir, try_generator=False):
+        # type: (File, bool) -> Generator[Union[Coordinate, str]]
         """ generate the missing tiles coordinates for low/high-zoom
             splitting
             :param try_generator: if set, it will to use the preset tile_coords_generator if that's available
         """
-        tmpdir = tempfile.mkdtemp()
-        try:
-            if try_generator and bool(self.tile_coords_generator):
-                print("[make_meta_tiles] generate missing tiles coords "
-                      "using customized generator")
-                return [coord for coord in self.tile_coords_generator.generate_tiles_coordinates([])]
-            else:
-                self.run_batch_job()
-                missing_meta_file = os.path.join(tmpdir, 'missing_meta.txt')
-                self.read_metas_to_file(missing_meta_file, compress=True)
-                with gzip.open(missing_meta_file, 'r') as fh:
-                    return [deserialize_coord(line) for line in fh]
-        finally:
-            shutil.rmtree(tmpdir)
+
+        if try_generator and bool(self.tile_coords_generator):
+            print("[make_meta_tiles] generate missing tiles coords "
+                    "using customized generator")
+            return self.tile_coords_generator.generate_tiles_coordinates([])
```

Comment:
Can you add a comment describing what the empty list is doing here?

Reply:
> Can you add a comment describing what the empty list is doing here?

the parameter is a zoom filter, I think I will make a default value for it so that caller don't need to pass an empty list to indicate no filter.